scripts/build.sh

```diff
@@ -430,11 +486,34 @@ function build_image {
         log_info "Added custom Maven args: ${MAVEN_ARGS_APPEND}"
     fi
 
-    # Validate Dockerfile exists
-    local dockerfile_path="$script_dir_path/../docker/osl.Dockerfile"
-    if [[ ! -f "$dockerfile_path" ]]; then
-        log_error "Dockerfile not found: $dockerfile_path"
-        return 20
+    # Set dockerfile path - use specified dockerfile or create temporary one from embedded content
+    local dockerfile_path
+    local temp_dockerfile_created=false
+    
+    if [[ -n "${args["dockerfile"]:-}" ]]; then
+        # Use specified dockerfile (handle both absolute and relative paths)
+        if [[ "${args["dockerfile"]}" = /* ]]; then
+            dockerfile_path="${args["dockerfile"]}"
+        else
+            dockerfile_path="$(realpath "${args["dockerfile"]}" 2>/dev/null || echo "${args["workflow-directory"]}/${args["dockerfile"]}")"
+        fi
+        log_info "Using specified dockerfile: ${args["dockerfile"]}"
+        
+        # Validate custom Dockerfile exists
+        if [[ ! -f "$dockerfile_path" ]]; then
+            log_error "Dockerfile not found: $dockerfile_path"
+            return 20
+        fi
+    else
+        # Create temporary dockerfile from embedded content
```

Comment:
I would love to see a log message similar to 
```
log_info "Using specified dockerfile: ${args["dockerfile"]}"
```
but for usage of the "default" embedded one

I know there is
```
log_info "Creating temporary dockerfile from embedded content: $dockerfile_path"
```
but I find the message less clear, you get the information in a transitive manner as the main info is the path of the dockerfile.

WDYT?